### ISSUE TITLE
Higher AP get worse visualization result

### ISSUE THREAD

**Issue by sleepat-11** [facebookresearch/Mask2Former#130]
hi, i trained Mask2Former on custom dataset using the following command:
```
python train_net.py     --config-file configs/coco/instance-segmentation/maskformer2_R50_bs16_50ep.yaml     --num-gpus 2     SOLVER.IMS_PER_BATCH 4     SOLVER.BASE_LR 0.00002
```
the AP result is 22.286

another model based on mask R-CNN is trained on the same dataset, 
AP got 14.959.
then i use two models to inference,here is the result:https://imgur.com/a/mRbUeQ0

left is the model based on Mask R-CNN,middle is Mask2Former(filtered the instance with confidence score), right is ground truth.

i have a few questions:
1.why the bbox predicted by Mask2Former enclose the whole image? (on other images, it predicted similar result)
2.is it possible to get higher ap with this visualization result?
Thanks in advance

**Comment by bowenc0221** [comment]
Looks like you visualized all predictions. Please try to only visualize predictions with high confidence scores.

**Comment by sleepat-11** [comment]
Thanks for your reply, the prediction showed above is already filtered with the confidence score  following the code in (https://github.com/facebookresearch/Mask2Former/issues/36)  
Most confidence scores are very low.For example,i `print(instances[index].scores)` and get 
```
tensor([0.0021])
tensor([0.0019])
tensor([0.0234])
tensor([0.0022])
tensor([0.0059])
tensor([0.0087])
tensor([0.0015])
tensor([0.0042])
```
here is other result https://imgur.com/a/19OTuJl, they are all filtered with confidence scores, it seems that the instance encloses the whole picture get high confidence score.and almost every bbox of instance are very big ,enclosed the image What is the reason of it?
my inference code
```
input_file = "output1/inference/valimg/rgb_0192.png"
im = cv2.imread(input_file)
outputs = predictor(im)
v = Visualizer(im[:, :, ::-1], coco_metadata, scale=1.2, instance_mode=ColorMode.IMAGE_BW)
# instance_result = v.draw_instance_predictions(outputs["instances"].to("cpu")).get_image()
instances = outputs["instances"].to("cpu") 
instances_ = Instances(instances.image_size)# <class 'detectron2.structures.instances.Instances'>
flag = False
for index in range(len(instances)):
        print(instances[index].scores)
        score = instances[index].scores[0]
        if score > 0.4: # confidence score
            if flag == False:
                instances_ = instances[index]
                flag = True
            else:
                instances_ = Instances.cat([instances_, instances[index]])
vis_output = v.draw_instance_predictions(predictions=instances_)
instance_result = vis_output.get_image()
cv2.imwrite('/mnt/data/test/predict_192.png',instance_result)
```

**Comment by sleepat-11** [comment]
> Looks like you visualized all predictions. Please try to only visualize predictions with high confidence scores.

Thanks for your reply, the prediction showed above is already filtered with the confidence score  following the code in (https://github.com/facebookresearch/Mask2Former/issues/36)  
Most confidence scores are very low.For example,i `print(instances[index].scores)` and get 
```
tensor([0.0021])
tensor([0.0019])
tensor([0.0234])
tensor([0.0022])
tensor([0.0059])
tensor([0.0087])
tensor([0.0015])
tensor([0.0042])
```
here is other result https://imgur.com/a/19OTuJl, they are all filtered with confidence scores, it seems that the instance encloses the whole picture get high confidence score.and almost every bbox of instance are very big ,enclosed the image What is the reason of it?
my inference code
```
input_file = "output1/inference/valimg/rgb_0192.png"
im = cv2.imread(input_file)
outputs = predictor(im)
v = Visualizer(im[:, :, ::-1], coco_metadata, scale=1.2, instance_mode=ColorMode.IMAGE_BW)
# instance_result = v.draw_instance_predictions(outputs["instances"].to("cpu")).get_image()
instances = outputs["instances"].to("cpu") 
instances_ = Instances(instances.image_size)# <class 'detectron2.structures.instances.Instances'>
flag = False
for index in range(len(instances)):
        print(instances[index].scores)
        score = instances[index].scores[0]
        if score > 0.4: # confidence score
            if flag == False:
                instances_ = instances[index]
                flag = True
            else:
                instances_ = Instances.cat([instances_, instances[index]])
vis_output = v.draw_instance_predictions(predictions=instances_)
instance_result = vis_output.get_image()
cv2.imwrite('/mnt/data/test/predict_192.png',instance_result)
```

**Comment by sleepat-11** [comment]
@bowenc0221 Thanks for your reply!
the prediction showed above is already filtered with the confidence score  following the code in (https://github.com/facebookresearch/Mask2Former/issues/36)  
Most confidence scores are very low.For example,i `print(instances[index].scores)` and get 
```
tensor([0.0021])
tensor([0.0019])
tensor([0.0234])
tensor([0.0022])
tensor([0.0059])
tensor([0.0087])
tensor([0.0015])
tensor([0.0042])
```
here is other result https://imgur.com/a/19OTuJl, they are all filtered with confidence scores, it seems that the instance encloses the whole picture get high confidence score.and almost every bbox of instance are very big ,enclosed the image What is the reason of it?
my inference code
```
input_file = "output1/inference/valimg/rgb_0192.png"
im = cv2.imread(input_file)
outputs = predictor(im)
v = Visualizer(im[:, :, ::-1], coco_metadata, scale=1.2, instance_mode=ColorMode.IMAGE_BW)
# instance_result = v.draw_instance_predictions(outputs["instances"].to("cpu")).get_image()
instances = outputs["instances"].to("cpu") 
instances_ = Instances(instances.image_size)# <class 'detectron2.structures.instances.Instances'>
flag = False
for index in range(len(instances)):
        print(instances[index].scores)
        score = instances[index].scores[0]
        if score > 0.4: # confidence score
            if flag == False:
                instances_ = instances[index]
                flag = True
            else:
                instances_ = Instances.cat([instances_, instances[index]])
vis_output = v.draw_instance_predictions(predictions=instances_)
instance_result = vis_output.get_image()
cv2.imwrite('/mnt/data/test/predict_192.png',instance_result)
```

**Comment by sleepat-11** [comment]
i solved the problem refer to [issue#95](https://github.com/facebookresearch/Mask2Former/issues/95#ref-issue-1278513267)
comment these two lines: https://github.com/facebookresearch/Mask2Former/blob/9b0651c6c1d5b3af2e6da0589b719c514ec0d69a/mask2former/modeling/meta_arch/mask_former_head.py#L33-L34

**Comment by Robotatron** [comment]
> i solved the problem refer to [issue#95](https://github.com/facebookresearch/Mask2Former/issues/95#ref-issue-1278513267) comment these two lines:
> 
> https://github.com/facebookresearch/Mask2Former/blob/9b0651c6c1d5b3af2e6da0589b719c514ec0d69a/mask2former/modeling/meta_arch/mask_former_head.py#L33-L34

Could you please explain what does this achieve and why does it help with inference? @sleepat-11
